libraries/node-core-library/src/FileSystem.ts

```diff
@@ -0,0 +1,432 @@
+   */
+  public static emptyFolder(folderPath: string): void {
+    fsx.emptyDirSync(folderPath);
+  }
+
+  // ===============
+  // FILE OPERATIONS
+  // ===============
+
+  /**
+   * Writes a text string to a file on disk, overwriting the file if it already exists.
+   * Behind the scenes it uses `fs.writeFileSync()`
+   * @param filePath - The absolute or relative path of the file.
+   * @param contents - The text that should be written to the file.
+   * @param options - Optional settings that can change the behavior. Type: `IWriteFileOptions`
+   */
+  public static writeFile(filePath: string, contents: string, options?: IWriteFileOptions): void {
+    options = {
+      ensureFolder: false,
+      convertLineEndings: NewlineConversion.None,
+      encoding: 'utf8',
+      ...options
+    };
+
+    if (options.ensureFolder) {
+      const folderPath: string = pathUtilities.dirname(filePath);
+      FileSystem.createFolder(folderPath);
+    }
+
+    contents = FileSystem._convertLineEndings(contents, options.convertLineEndings);
+
+    fsx.writeFileSync(filePath, contents, { encoding: options.encoding });
+  }
+
+  /**
+   * Reads the contents of a file into a string.
+   * Behind the scenes it uses `fs.readFileSync()`.
+   * @param filePath - The path of the file to be read.
+   * @param options - Optional settings that can change the behavior. Type: `IReadFileOptions`
+   */
+  public static readFile(filePath: string, options?: IReadFileOptions): string {
+    options = {
+      encoding: 'utf8',
+      convertLineEndings: NewlineConversion.None,
+      ...options
+    };
+
+    const contents: string = FileSystem.readFileToBuffer(filePath).toString(options.encoding);
+    return FileSystem._convertLineEndings(contents, options.convertLineEndings);
+  }
+
+  public static readFileToBuffer(filePath: string): Buffer {
+    return fsx.readFileSync(filePath);
+  }
+
+  /**
+   * Copies a file from one location to another.
+   * By default, destinationPath is overwritten if it already exists.
+   * Behind the scenes it uses `fs.copyFileSync()`.
+   * @param sourcePath - The absolute or relative path to the source file to be copied.
+   * @param destinationPath - The absolute or relative path to the new copy that will be created.
+   */
+  public static copyFile(sourcePath: string, destinationPath: string): void {
+    fsx.copySync(sourcePath, destinationPath);
+  }
+
+  /**
+   * Deletes a file. Can optionally throw if the file doesn't exist.
+   * Behind the scenes it uses `fs.unlinkSync()`.
+   * @param filePath - The absolute or relative path to the file that should be deleted.
+   * @param options - Optional settings that can change the behavior. Type: `IDeleteFileOptions`
+   */
+  public static deleteFile(filePath: string, options?: IDeleteFileOptions): void {
+    options = {
+      throwIfNotExists: false,
+      ...options
+    };
+
+    if (options.throwIfNotExists || FileSystem.exists(filePath)) {
+      fsx.unlinkSync(filePath);
```

Comment:
Here we have a race condition: The person said NOT to throw if it exists, but it the file could have been created immediately after we checked for existence.

I would suggest to add a try-catch block for that edge case (while still keeping the check to avoid unexceptional exceptions).